fastlane/lib/fastlane/action.rb

```diff
@@ -118,9 +118,24 @@ def self.method_missing(method_sym, *arguments, &_block)
       UI.user_error!("To call another action from an action use `OtherAction.#{method_sym}` instead")
     end
 
-    # When shelling out from the actoin, should we use `bundle exec`?
+    # When shelling out from the action, should we use `bundle exec`?
     def self.shell_out_should_use_bundle_exec?
-      return File.exist?('Gemfile') && !Helper.contained_fastlane?
+      unless Helper.contained_fastlane?
+        given = ENV["BUNDLE_GEMFILE"]
+        return true if given && !given.empty?
+
+        previous = nil
+        current  = File.expand_path(Pathname.pwd).untaint
+        until !File.directory?(current) || current == previous
+          filename = File.join(current, "Gemfile")
```

Comment:
Great point, thanks @segiddins, @jcayzac there is a method in `bundler` that allows you to easily get the name for the Gemfile in the current directory, let me know if you need any help 👍